pkg/apis/duck/v1/delivery_types.go

```diff
@@ -98,9 +98,8 @@ func (ds *DeliverySpec) Validate(ctx context.Context) *apis.FieldError {
 
 	if ds.Timeout != nil {
 		if feature.FromContext(ctx).IsEnabled(feature.DeliveryTimeout) {
-			t, te := period.Parse(*ds.Timeout)
-			if te != nil || t.IsZero() {
-				errs = errs.Also(apis.ErrInvalidValue(*ds.Timeout, "timeout"))
+			if _, te := isoduration.Parse(*ds.Timeout, isoduration.IsPositive); te != nil {
```

Comment:
Two questions:

Would this read better as `isoduration.Parse(ds.Timeout).AssertPositive()`?

Are we discovering the consequences of not following https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#units, or is this due to a need for high-precision timeout durations?

I'm slightly concerned about the tradeoffs of human and machine readability with ISO durations, but I don't have a specific better suggestion.

I will point out that timeout durations less than 1s will start to run into TCP retry limitations, and might not be as useful as they first appear, especially if retry interval is linked to timeout.